src/components/Editor/TextEditor.tsx

```diff
@@ -42,52 +33,72 @@ const TextEditor = (): JSX.Element => {
       setCurrDesc("");
     }
   }, [selectedAction, actionList]);
-  return isExpanded ? (
-    <TextDiv>
-      <Header>
-        <IconButton
-          sx={{ color: "white" }}
-          onClick={() => {
-            setExpanded((value) => !value);
-          }}
-        >
-          <ExpandIcon sx={{ fontSize: 30, ml: "0.5em" }} />
-        </IconButton>
-        <IconButton
-          sx={{ color: "white" }}
-          onClick={() => setDescription(currDesc)}
-        >
-          <SaveAltIcon sx={{ fontSize: 30 }} />
-        </IconButton>
-      </Header>
-      <Box
-        sx={{
-          m: "1em",
-          mt: "0.5em",
-          mb: "1.2em",
-          mr: "1.3em",
-          alignItems: "center",
-          height: "100%",
-        }}
-      >
-        <textarea
-          style={{
-            background: "white",
-            borderRadius: "5px",
-            width: "100%",
+  return (
+    <Rnd
+      cancel=".noDrag"
+      minWidth="200"
+      minHeight="150"
+      maxWidth="400"
+      maxHeight="250"
+    >
+      {" "}
+      <TextDiv>
+        <Header>
+          <IconButton
+            className={styles.sidebarItem}
+            onClick={() => {
+              setDescription(currDesc);
+            }}
+          >
+            <SaveIcon
+              sx={{
+                fontSize: 25,
+                marginRight: "0.3rem",
+              }}
+            />
+          </IconButton>
+          <IconButton
+            className={styles.sidebarItem}
+            onClick={() => {
+              toggleTextBox();
+            }}
+          >
+            <CancelIcon
+              sx={{
+                fontSize: 25,
+                marginRight: "0.3rem",
+              }}
+            />
+          </IconButton>
+        </Header>
+        <Box
+          className="noDrag"
+          sx={{
+            position: "relative",
+            ml: "1em",
+            mr: "1.4em",
+            mt: "0.5em",
+            mb: "1.2em",
+            alignItems: "center",
             height: "100%",
-            resize: "none",
-            fontSize: "1.6em",
           }}
-          value={currDesc}
-          onChange={(e) => setCurrDesc(e.target.value)}
-        />
-      </Box>
-    </TextDiv>
-  ) : (
-    <DefaultButton onClick={() => setExpanded((value) => !value)}>
-      ADD/EDIT DESCRIPTION
-    </DefaultButton>
+        >
+          <textarea
+            className="noDrag"
+            style={{
+              background: "white",
+              borderRadius: "5px",
+              width: "100%",
+              height: "100%",
+              resize: "none",
+              fontSize: "1.3rem",
+            }}
+            value={currDesc}
+            onChange={(e) => setCurrDesc(e.target.value)}
+          />
+        </Box>
+      </TextDiv>
+    </Rnd>
```

Comment:
I was thinking if it's possible to extract the style for each div/tag to make it easier to read and change if needed in the future.

Reply:
that's a very good point. I'm working on another PR for enhancement. Will fix this